src/sentry/coreapi.py

```diff
@@ -369,12 +380,17 @@ def parse_client_as_sdk(self, value):
         }
 
     def should_filter(self, project, data, ip_address=None):
-        # TODO(dcramer): read filters from options such as:
-        # - ignore errors from spiders/bots
-        # - ignore errors from legacy browsers
         if ip_address and not is_valid_ip(ip_address, project):
             return True
 
+        release = data.get('release')
+        if release and not is_valid_release(release, project):
+            return True
+
+        error_message = data.get('sentry.interfaces.Message', {}).get('message')
+        if error_message and not is_valid_error_message(error_message, project):
```

Comment:
https://github.com/getsentry/sentry/blob/master/src/sentry/interfaces/message.py#L24-L37

`message` is the only field that's (ostensibly) required so I'd keep it as a fallback but I'd check `formatted` first and use that if it's available.

Also be aware that you could potentially hit this branch (:grimacing:) and should probably handle it gracefully: https://github.com/getsentry/sentry/blob/master/src/sentry/interfaces/message.py#L47-L50 (This where I usually put some jaded comment about type validation but not this time)